src/components/badge/badge.test.tsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import Badge from "./badge.component";
+
+const renderComponent = (props = {}) =>
+  render(<Badge {...props}>Button</Badge>);
+
+describe("Badge", () => {
+  it("should render correctly if counter is a number", async () => {
+    renderComponent({ counter: 1 });
+
+    expect(screen.getByText("1")).toBeVisible();
+  });
+
+  it("should render correctly if counter is a string", async () => {
+    renderComponent({ counter: "1" });
+
+    expect(screen.getByText("1")).toBeVisible();
+  });
+
+  it("should render number when counter is between 1 and 99", async () => {
+    renderComponent({ counter: 50 });
+
+    expect(screen.getByText("50")).toBeVisible();
+  });
+
+  it("should render number `99` when counter is higher than 99", async () => {
+    renderComponent({ counter: 101 });
+
+    expect(screen.getByText("99")).toBeVisible();
+  });
+
+  it("should not render badge when counter is `0`", async () => {
+    renderComponent({ counter: 0, onClick: () => {} });
+
+    expect(screen.queryByText("0")).not.toBeInTheDocument();
+    expect(screen.queryByRole("button")).not.toBeInTheDocument();
```

Comment:
comment: I don't think you need both assertions here, the first one suffices imo